src/receive_imf.rs

```diff
@@ -1482,6 +1482,15 @@ async fn add_parts(
                 Reaction::from(reaction_str.as_str()),
             )
             .await?;
+            if mime_parser.incoming && in_fresh {
+                if let Some((msg_id, _)) = rfc724_mid_exists(context, mime_in_reply_to).await? {
+                    context.emit_event(EventType::IncomingReaction {
```

Comment:
But this emits an event for any reaction, not only to one's own sent message

Reply:
you're right, the condition is not sufficient; `in_fresh` was also wrong (fixed that and added a little test)

Reply:
> But this emits an event for any reaction, not only to one's own sent message

should be be fixed by the last commit. 

not sure, however, how to test the absence of an event - the event loop should be empty, maybe test for that? @iequidoo do you know, how that can be done or even have better ideas?

Reply:
You can use `EventType::Test` as a fence, just emit it and check that the next event is `Test`. That means there are no other events in the queue. Almost all the core code emit events "synchronously", there should be no parallel tasks continuing emitting events, so i think such a check is fine

Reply:
hm, this does not work as there are lots of other events in queue.

instead, a function that checks if a given EventType is not in event queue might be helpful, i tried to add the following to `EventTracker`, but that did not work:

```
    /// Consumes events and errors out if the given event type is found.
    pub async fn assert_not_emitted(&self, event_type: EventType) {
        loop {
            if let Ok(event) = self.try_recv() {
                assert_ne!(event.typ, event_type);
            } else {
                return;
            }
        }
    }
```

calling is in reaction.rs test as:

```
    async fn expect_no_incoming_reactions_event(t: &TestContext){
        t.evtracker.assert_not_emitted(EventType::IncomingReaction {
            contact_id: Default::default(),
            msg_id: Default::default(),
            reaction: "".to_string(),
        }).await
    }
```

anyways, that should not block this PR, we're nowhere testing for event not emitted yet :) but if someone has an fix or and idea, we can put that in (or in a subsequent PR then)